### ISSUE TITLE
Allow plugin timeout to be customized

### ISSUE THREAD

**Issue by andy9775** [victorcoder/dkron#507]
I'd like to propose the ability to customize the time that dkron waits for a plugin to start. From my quick look, it seems as though it can be customized [here](https://github.com/gazoomobile/dkron-executor-rabbitmq) using `config.StartTimeout`. I'm blocked by #500 for the time being so I can't full test it out.

The reasoning behind this is that I've forked and updated a [RabbitMQ executor](https://github.com/gazoomobile/dkron-executor-rabbitmq) that I've found but ran into a small issue. If the RabbitMQ service isn't booted and setup before starting dkron, the executor fails to work as the drkon boot process calls the executor which attempts to connect to the rabbit service, fails and returns an error. I could of course attempt to reconnect when the plugin runs but I think it makes sense to delay the dkron startup process rather than having the executor check each time it's used.

We could keep the config optional (it defaults to 1 minute if not provided) with the ability for the user to config it themselves. Config options could be:
- the timeout must be in seconds
- the timeout must a valid go time duration string ("5m" for example)
We can use either an [env variable or the config file](https://dkron.io/basics/configuration/) to specify this configuration.

**Comment by victorcoder** [comment]
If I understand correctly that won't fix the root of the problem. The plugin doesn't have to connect on startup but on the first usage. RabbitMQ uses persistent connections, it should capture the error in case the connection is closed, not crash.

What do you think?

**Comment by andy9775** [comment]
Ya, that is valid and it would work as well.  However connecting and setting up on dkron boot makes more sense rather than having the connection occur at some later time. This way the setup of the executor is tied to the setup of dkron. 

By having the connection occur later, we delay knowing if the executor connected successfully or not.

**Comment by victorcoder** [comment]
In any case, the plugin should not crash but capture the error and log it to give the user the opportunity to fix it.

**Comment by andy9775** [comment]
If the fix is "wait for rabbit to boot and then try again" this should be done automatically rather than rebooting dkron manually by the user. In the current scenario, dkron then ends up being dependent on rabbit already running (or any other executor which depends on a third party service) before being able to start up.

While dkron does wait already, 1 minutes seems to be to short (for rabbit specifically). Customization of the timeout value gives some leeway for things to start up by themselves in (relatively) any order.

**Comment by vcastellm** [comment]
Not enough interest
